### PR TITLE
Feature/76 draw game

### PR DESCRIPTION
**Pull request by sevap8**

### CODE REVIEW

**Comment by sevap8** [comment]
Hello everyone, please see how there will be time, I didn’t finish a few things there, but there’s a general logic, I’ll do the cop style and corrections in the evening by sending it, because I have to go to work.
I will be glad of your comment

File: HW3/Seva_Popov/Draw-Game/Draw-Game/GameLogics.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            {
+                this.board.Create(board);
+                Draw draw = delegate (IBoard board) { };
+                Console.WriteLine(phraseProvider.GetPhrase("1"));
+                Console.WriteLine(phraseProvider.GetPhrase("2"));
+                Console.WriteLine(phraseProvider.GetPhrase("3"));
+                Console.WriteLine(phraseProvider.GetPhrase("4"));
+                string st = reader.ReadLine();
+
+                if (st.Equals("1"))
```
Comment (MilRoad) [inline HW3/Seva_Popov/Draw-Game/Draw-Game/GameLogics.cs]: Your numbers don't match with your lines.
For example, you write information that if I click 1, I'll draw simple point, but I see horizontal line.
Just swap it :)

File: HW3/Seva_Popov/Draw-Game/Draw-Game/GameLogics.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    draw += this.board.SimpleDot;
+                    draw(board);
+                }
+
+                if (st.Equals("4"))
+                {
+                    draw += this.board.AnotherShape;
+                    draw(board);
+                }
+                if (st.Equals(settingsProvider.GetGameSettings().ExitCode))
```
Comment (MilRoad) [inline HW3/Seva_Popov/Draw-Game/Draw-Game/GameLogics.cs]: You need to use ToLowerInvariant() for user input data when you compare it with exit code.

File: HW3/Seva_Popov/Draw-Game/Draw-Game/Interfaces/IBoard.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    using System.Collections.Generic;
+    using System.Text;
+
+   public interface IBoard
+    {
+         int SizeX { get; set; }
+         int SizeY { get; set; }
+         void WriteAt(string s, int x, int y);
+         void Create(IBoard board);
+         void AnotherShape(IBoard board);
```
Comment (MilRoad) [inline HW3/Seva_Popov/Draw-Game/Draw-Game/Interfaces/IBoard.cs]: Maybe you should add word "Draw" to the names of your methods to say more clearly what they do.

File: HW3/Seva_Popov/Draw-Game/Draw-Game/Interfaces/IOutputWriter .cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Draw_Game.Interfaces
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    public interface IOutputWriter
+    {
+        void Write(string message);
+        void СlearСonsole();
```
Comment (MilRoad) [inline HW3/Seva_Popov/Draw-Game/Draw-Game/Interfaces/IOutputWriter .cs]: I don't think that this method must be here. It's better to rename it or rename and move it to IBoard. 
It's interface, output device may be not a console.

File: HW3/Seva_Popov/Draw-Game/Draw-Game/Components/InputReader.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Draw_Game.Components
+{
+    using Draw_Game.Interfaces;
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    public class InputReader : IInputReader
```
Comment (bartimys1) [inline HW3/Seva_Popov/Draw-Game/Draw-Game/Components/InputReader.cs]: maybe it's worth combining input and output

File: HW3/Seva_Popov/Draw-Game/Draw-Game/Components/Board.cs [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        public static int origRow;
+        public static int origCol;
+
+        public void WriteAt(string s, int x, int y)
+        {
+            try
+            {
+                Console.SetCursorPosition(origCol + x, origRow + y);
```
Comment (bartimys1) [inline HW3/Seva_Popov/Draw-Game/Draw-Game/Components/Board.cs]: u have OutputReader, why u use System.Console?

File: HW3/Seva_Popov/Draw-Game/Draw-Game/Components/InputReader.cs (hunk 5 above)
Comment (sevap8) [reply HW3/Seva_Popov/Draw-Game/Draw-Game/Components/InputReader.cs]: I think this is a controversial issue, and if we would have a lot of information to enter outside, I like it when they are separated

File: HW3/Seva_Popov/Draw-Game/Draw-Game/Components/Board.cs (hunk 6 above)
Comment (sevap8) [reply HW3/Seva_Popov/Draw-Game/Draw-Game/Components/Board.cs]: yes, I will correct now

File: HW3/Seva_Popov/Draw-Game/Draw-Game/Interfaces/IOutputWriter .cs (hunk 4 above)
Comment (bartimys1) [reply HW3/Seva_Popov/Draw-Game/Draw-Game/Interfaces/IOutputWriter .cs]: just rename. don't need to move it to board

File: HW3/Seva_Popov/Draw-Game/Draw-Game/Components/InputReader.cs (hunk 5 above)
Comment (bartimys1) [reply HW3/Seva_Popov/Draw-Game/Draw-Game/Components/InputReader.cs]: maybe u are right. I did not think about it

File: HW3/Seva_Popov/Draw-Game/Draw-Game/Components/Board.cs (hunk 6 above)
Comment (sevap8) [reply HW3/Seva_Popov/Draw-Game/Draw-Game/Components/Board.cs]: fix

File: HW3/Seva_Popov/Draw-Game/Draw-Game/GameSettings.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    public class GameSettings
+    {
+        public int Length { get; set; }
+
+        public int Width { get; set; }
```
Comment (Foxtra) [inline HW3/Seva_Popov/Draw-Game/Draw-Game/GameSettings.cs]: You have no any check for Length and Width of your board, for example for this values dot is overwriten by "AnotherShape".
изображение

**Pull request merged**